### DISCUSSION TITLE
Qobuz

### DISCUSSION THREAD

**Discussion by marcelveldt** [music-assistant/support#817] [Music providers]
Music Assistant has full support for [Qobuz](https://www.qobuz.com/)

Documentation https://music-assistant.io/music-providers/qobuz/

**Comment by michelsup** [comment]
Hello Marcel. Thank you very much for your work. Is the Qobuz's radio feature is in the roadmap pipe somehow?

**Reply by marcelveldt** [reply]
Well, problem is that Qobuz closed their API nowadays and removed the API documentation. 
Lucky for us we got on board before the closing so it will keep working but for newer endpoints I will have to reverse engineer it from their web client. So yeah, this will come at one point but its a bit low on the prio list for me, hoping someone gets bored and can do soe of the plumbing already (figuring out the endpoints).

**Reply by junado** [reply]
I'd love to have a look at the radio feature. I've been reverse engineering the Qobuz API the past few weeks to pull the Qobuz curated playlists into Home Assistant to play through MA. I had a quicklook at how the radio feature works on the web player and it appears to select the next song based on the current playing song and does not build a full playlist. Is this something that could be supported by MA ? If so, I'll start digging.

Edit: further investigation reveals that it can generate up to the next 50 songs from a number of listened track IDs and their corresponding artists, genre and label ids. I'll keep digging - this is a feature I'd love to get to work. I'll move the discussion to the Discord channel if it makes more sense.

**Reply by richardwonka** [reply]
Yes, please, I have been using radio features extensively and it's something I miss with Qobuz on MA

**Comment by ncd7** [comment]
Since YTM authentication no longer works for me, I was digging through other available providers and realized Qobuz is on the list. Thanks for implementing it!
I have a paid qobuz membership and just tried logging into it on MA and it timed out. I tried a few times to no avail. 
Does this still work? I know it's more of a niche service so not sure how actively it is supported.
Thanks.

**Reply by OzGav** [reply]
Yes it does still work.

**Reply by ncd7** [reply]
Then I must be one unlucky guy. 
It doesn't work for me. I put in my username (which is in the form of an email, could that be the issue) and password, the it just spins forever and fails. 
FYI I've verified I can log into Qobuz with those credentials fine. 
This is what I see
![image](https://github.com/user-attachments/assets/3cba399f-660a-4a00-a82e-9cecfe69b9c0)

**Reply by OzGav** [reply]
Try a different browser?

**Reply by marcelveldt** [reply]
Qobuz should still working fine, I use it everyday.
Maybe special network config (like pihole, adguard) or whatever on your network or any other reason why the HA/MA server can not contact qobuz ?

**Reply by ncd7** [reply]
Thanks for the suggestion, I do not have any adblocking addons nor anything at the network level like pihole.

**Reply by ncd7** [reply]
Ah, I do have a network level VPN specifically for my IoT network on which HA and MA sit. Let's see if this might be it! Thanks for the suggestion.

**Comment by michelsup** [comment]
You need to disable any ad blocker, privacy addon, etc. for the credentials to work.

**Reply by ncd7** [reply]
I do not believe in adblockers (personally! everyone is free to choose as they wish, of course) so I do not have any.

**Comment by richardwonka** [comment]
While I can get my sonos speakers to play from the Qobuz source, I am having trouble finding the playlists Qobuz generates (e.g. DailyQ, WeeklyQ, etc..) 

Any pointers are appreciated

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
